diff -ruN darknet/src/image.c /home/lennon/darknet_ros/darknet/src/image.c
--- darknet/src/image.c	2018-08-20 10:25:12.334532970 +0800
+++ /home/lennon/darknet_ros/darknet/src/image.c	2018-08-06 15:08:24.048390000 +0800
@@ -591,7 +591,23 @@
 
 void ipl_into_image(IplImage* src, image im)
 {
-    unsigned char *data = (unsigned char *)src->imageData;
+    // unsigned char *data = (unsigned char *)src->imageData;
+    // int h = src->height;
+    // int w = src->width;
+    // int c = src->nChannels;
+    // int step = src->widthStep;
+    // int i, j, k;
+
+    // for(i = 0; i < h; ++i){
+    //     for(k= 0; k < c; ++k){
+    //         for(j = 0; j < w; ++j){
+    //             im.data[k*w*h + i*w + j] = data[i*step + j*c + k]/255.;
+    //         }
+    //     }
+    // }
+    IplImage* temp = cvCloneImage(src);
+    //unsigned char *data = (unsigned char *)src->imageData;
+    unsigned char *data = (unsigned char *)temp->imageData;
     int h = src->height;
     int w = src->width;
     int c = src->nChannels;
@@ -601,10 +617,11 @@
     for(i = 0; i < h; ++i){
         for(k= 0; k < c; ++k){
             for(j = 0; j < w; ++j){
-                im.data[k*w*h + i*w + j] = data[i*step + j*c + k]/255.;
+                im.data[k*w*h + i*w + j] = data[i*step + j*c + k]/255.0;
             }
         }
     }
+    cvReleaseImage(&temp);
 }
 
 image ipl_to_image(IplImage* src)
